modules/atomic-widgets/elements/atomic-svg/atomic-svg.php

```diff
@@ -31,82 +30,73 @@ public function get_icon() {
 
 	protected function render() {
 		$settings = $this->get_atomic_settings();
+		$svg_url = $settings['svg']['src'] ?? '';
 
-		if ( ! isset( $settings['svg'] ) ) {
+		if ( ! $svg_url ) {
+			// we need this line in order to render the svg, otherwise it will be rendered as a string
+			printf( '%s', self::get_default_svg() ); // phpcs:ignore WordPress.Security.EscapeOutput.OutputNotEscaped
 			return;
 		}
 
-		$svg_data = $settings['svg'];
-
-		$svg_content = isset( $svg_data['url'] ) ? $svg_data['url'] : '';
-
-		if ( strpos( $svg_content, self::SVG_ENCODING ) === 0 ) {
-			$svg_content = urldecode( substr( $svg_content, self::SVG_START_INDEX ) );
-		}
+		$svg_content = ( strpos( $svg_url, self::SVG_PREFIX ) === 0 )
+			? urldecode( substr( $svg_url, strlen( self::SVG_PREFIX ) ) )
+			: wp_remote_retrieve_body( wp_safe_remote_get( $svg_url ) );
 
 		$svg = new \WP_HTML_Tag_Processor( $svg_content );
 
-		if ( ! $svg->next_tag() ) {
-			return;
+		if ( $svg->next_tag( 'svg' ) ) {
+			$this->set_svg_attributes( $svg, $settings );
 		}
 
-		$this->set_svg_attributes( $svg, $settings );
-
-		$svg->add_class( $settings['classes'] ?? '' );
+		while ( $svg->next_tag( [ 'path', 'rect', 'circle', 'g' ] ) ) {
+			$svg->remove_attribute( 'fill' );
+		}
 
 		$valid_svg = ( new Svg_Sanitizer() )->sanitize( $svg->get_updated_html() );
 
 		// we need this line in order to render the svg, otherwise it will be rendered as a string
-		echo ( false === $valid_svg ) ? '' : $valid_svg; //phpcs:ignore WordPress.Security.EscapeOutput.OutputNotEscaped
+		printf( '%s', $valid_svg ? $valid_svg : self::get_default_svg() ); // phpcs:ignore WordPress.Security.EscapeOutput.OutputNotEscaped
 	}
 
-	private function set_svg_attributes( $svg, $settings ) {
+	private function set_svg_attributes( \WP_HTML_Tag_Processor $svg, $settings ) {
 		$svg->set_attribute( 'fill', $settings['color'] ?? 'currentColor' );
-		$svg->set_attribute( 'width', $settings['width'] ?? '100%' );
-		$svg->set_attribute( 'height', $settings['height'] ?? '100%' );
+		$svg->set_attribute( 'width', $settings['width'] ?? '100px' );
+		$svg->set_attribute( 'height', $settings['height'] ?? '100px' );
+		$svg->add_class( $settings['classes'] ?? '' );
 	}
 
 	protected function define_atomic_controls(): array {
-		$content_section = Section::make()
-			->set_label( esc_html__( 'Content', 'elementor' ) )
-			->set_items( [
-				Svg_Control::bind_to( 'svg' ),
-			] );
-
 		return [
-			$content_section,
+			Section::make()
+				->set_label( esc_html__( 'Content', 'elementor' ) )
+				->set_items( [ Svg_Control::bind_to( 'svg' ) ] ),
```

Comment:
Where is the check if the user actually can upload an SVG? Is it in the frontend